content/source/docs/enterprise/api/ptfe_settings.html.md

```diff
@@ -0,0 +1,489 @@
+---
+layout: enterprise2
+page_title: "Private Terraform Settings - API Docs - Terraform Enterprise"
+sidebar_current: "docs-enterprise2-api-private-terraform-settings"
+---
+
+# Private Terraform Settings API
+
+-> **Note**: These API endpoints are in beta and are subject to change. Additionally, these endpoints are only available in Private Terraform instances and only accessible by site administrators.
+
+## List General Settings
+`GET /api/v2/admin/settings/general`
+
+-> **Note:** This endpoint cannot be accessed with [organization tokens](../users-teams-organizations/service-accounts.html#organization-service-accounts). You must access it with a [user token](../users-teams-organizations/users.html#api-tokens) which has the `site-admins` permission.
+
+Status  | Response                                         | Reason
+--------|--------------------------------------------------|-------
+[200][] | [JSON API document][] (`type: "settings"`) | Successfully listed General settings
+[404][] | [JSON API error object][]                    | User unauthorized to perform action
+
+[200]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/200
+[404]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/404
+[JSON API document]: https://www.terraform.io/docs/enterprise/api/index.html#json-api-documents
+[JSON API error object]: http://jsonapi.org/format/#error-objects
+
+### Sample Request
+
+```shell
+curl \
+  --header "Authorization: Bearer $ATLAS_TOKEN" \
+  --header "Content-Type: application/vnd.api+json" \
+  --request GET \
+  https://app.terraform.io/api/v2/admin/settings/general
+```
+
+### Sample Response
+
+```json
+{
+  "data": {
+    "id": "general-settings",
+    "type": "settings",
```

Comment:
Based on a quick internal spike, the implementation is going to be easier with a couple of changes here:

- Let's make the `type` for each setting follow the pattern `settings/general`, `settings/smtp`, etc. This matches the path and the model name in our UI;
- Then for reduced redundancy, we can remove `-settings` from the `id`, i.e. `general`, `smtp`. This is for reduced redundancy.